source/Halibut.Tests/PublicSurfaceAreaFixture.ThePublicSurfaceAreaShouldNotRegress.NETFramework.approved.cs

```diff
@@ -17,8 +17,10 @@ public class DataStream : IEquatable<Halibut.DataStream>
     {
         public DataStream() { }
         public DataStream(long length, Action<Stream> writer) { }
+        public DataStream(Guid guid, Func<Halibut.DataStream> delayedStreamCreation) { }
```

Comment:
similar to previous comment: looks like we're passing a `DataStream` to a `DataStream` ctor here, which makes me wonder if this new requirement is being introduced at the right abstraction. Could we achieve something similar by making the calling code accept a `Lazy<DataStream>` instead?